game-core/src/main/java/games/strategy/triplea/ui/PlacePanel.java

```diff
@@ -94,13 +107,106 @@ public void territorySelected(final Territory territory, final MouseDetails e) {
         }
       };
 
-  PlacePanel(final GameData data, final MapPanel map, final TripleAFrame frame) {
+  enum Mode {
+    UNITS_TO_PLACE_VIEW_ATTACHED,
+    UNITS_TO_PLACE_VIEW_DETACHED
+  }
+
+  PlacePanel(final GameData data, final MapPanel map, final Mode mode, final TripleAFrame frame) {
     super(data, map, frame);
+    this.mode = mode;
     undoableMovesPanel = new UndoablePlacementsPanel(this);
-    unitsToPlace = new SimpleUnitPanel(map.getUiContext());
+    if (mode == Mode.UNITS_TO_PLACE_VIEW_DETACHED) {
+      unitsToPlacePanel =
+          new SimpleUnitPanel(
+              map.getUiContext(), SimpleUnitPanel.Style.SMALL_ICONS_WRAPPED_WITH_LABEL_WHEN_EMPTY);
+      unitsToPlacePanel.setBorder(BorderFactory.createTitledBorder("Units to Place"));
+      unitsToPlacePanel.setVisible(false);
+      data.addGameDataEventListener(GameDataEvent.GAME_STEP_CHANGED, this::updateStep);
+    } else {
+      unitsToPlacePanel = new SimpleUnitPanel(map.getUiContext());
+    }
     leftToPlaceLabel.setText("Units left to place:");
   }
 
+  private void updateStep() {
+    final Collection<UnitCategory> unitsToPlace;
+    final GameData data = getData();
+    data.acquireReadLock();
+    try {
+      final GameStep step = data.getSequence().getStep();
+      if (step == null) {
+        return;
+      }
+      // Note: This doesn't use getCurrentPlayer() as that may not be updated yet.
+      final PlayerId player = step.getPlayerId();
+      unitsToPlace = getUpdatedUnitsToPlace(lastPlayer, player, step);
+      lastPlayer = player;
+    } finally {
+      data.releaseReadLock();
+    }
+
+    SwingUtilities.invokeLater(
+        () -> {
+          JSplitPane splitPane = null;
+          final Component parent = unitsToPlacePanel.getParent();
+          if (parent instanceof JSplitPane) {
```

Comment:
I'm curious which cases will `parent` not be an instance of `JSplitPane`? Can we perhaps make this more deterministic? Isn't `unitsToPlacePanel` always in a split pane? Could we perhaps just keep a reference to the split pane instead of relying on `getParent()`?

Reply:
I believe it's always an instance of JSplitPane in this method. Sure, I can simplify. The reason I had the extra logic is to make it so this class has less assumptions about how the detached panel works - i.e. it does the right thing in a split pane, but since the decision to put into a split pane is made outside this class, it doesn't break if something else is done.

But given the current use is to always put in a split pane, we could make the code simpler and always rely on that.